src/main/webapp/resources/js/pages/pipelines/pipeline.launch.js

```diff
@@ -364,9 +382,58 @@
 		svc.resetCurrentSelection = function () {
 			selectedParameters.currentSettings = ng.copy(selectedParameters.defaultSettings);
 		}
-  }
+    }
 
-  function FileUploadCtrl($rootScope, Upload) {
+    /**
+     * Service for handling Tool Data Tables.
+     */
+    function ToolDataTableService() {
+        var svc = this;
+
+        // Check to see if there are any tool data tables, if not put a default
+        if(page.pipeline.toolDataTables == null) {
+            page.pipeline.toolDataTables = [{
+                id: "no_tool_data_tables",
+                label: "",
+                parameters: []
+            }];
+        }
+
+        var settings = {};
+        if (page.pipeline.toolDataTables != null) {
+            settings["currentSettings"] = {};
+            settings["availableSettings"] = {};
+            for (var toolDataTable of page.pipeline.toolDataTables) {
```

Comment:
The `for...of` iterator is not supported in Internet Explorer [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).  We transpile a lot of the javascript code using babel, but unfortunately this file has not yet been updated to do that.  Sorry.